core/DataTable/Filter/ColumnDelete.php

```diff
@@ -161,4 +145,46 @@ public function filter($table)
 
         return $table;
     }
+
+    /**
+     * @param $table
+     * @return array
+     */
+    protected function removeColumnsFromTable(&$table)
+    {
+        if(!$this->isArrayAccess($table)) {
```

Comment:
It's not weird, it's the advantage of "interfaces" etc being able to treat them the same. It is now throwing notices probably since you start iterating nested on possible scalar values etc maybe